core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/AbstractFileBatchSource.java

```diff
@@ -96,7 +96,11 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
     if (!config.containsMacro("timeTable") && config.timeTable != null) {
       pipelineConfigurer.createDataset(config.timeTable, KeyValueTable.class, DatasetProperties.EMPTY);
     }
-    pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    if (config.getSchema() != null) {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(config.getSchema());
+    } else {
+      pipelineConfigurer.getStageConfigurer().setOutputSchema(DEFAULT_SCHEMA);
```

Comment:
For setting the actual output schema in the case of when the schema is not defined, we will only be able to get the schema once the record reader has been initialized and we start reading